### PR TITLE
Save selected search tab

### LINKED ISSUE #45278: rustdoc search engine should stay in the same tab
After selecting a category changing the search query will reset the tab selection.

cc @GuillaumeGomez

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #45278.

r? @rust-lang/docs

### CODE REVIEW

**Comment by dns2utf8** [comment]
Very cool 👍

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -816,9 +823,10 @@
             output = '<h1>Results for ' + escape(query.query) +
                 (query.type ? ' (type: ' + escape(query.type) + ')' : '') + '</h1>' +
                 '<div id="titles">' +
-                '<div class="selected">Types/modules</div>' +
-                '<div>As parameters</div>' +
-                '<div>As return value</div></div><div id="results">';
+                `<div${currentTab === 0 ? ' class="selected"' : ""}>Types/modules</div>` +
+                `<div${currentTab === 1 ? ' class="selected"' : ""}>As parameters</div>` +
+                `<div${currentTab === 2 ? ' class="selected"' : ""}>As return value</div>` +
```
Comment (kennytm) [inline src/librustdoc/html/static/main.js]: I don't think we should use template literals?

* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Browser_compatibility
* https://caniuse.com/#feat=template-literals

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: Excellent point!

**Comment by GuillaumeGomez** [comment]
Updated.

**Review by dns2utf8 — APPROVED** [review]
:+1:

**Comment by QuietMisdreavus** [comment]
@bors r+

I wonder how long it'll take until someone asks this to be saved to local storage. `>_>` Maybe that can be folded into an eventual settings page PR for https://github.com/rust-lang/rust/issues/18167.

**Comment by GuillaumeGomez** [comment]
@QuietMisdreavus: Sounds like a strange idea... It's just a `tab` away. :p

**Pull request merged**